extensions/amp-story/0.1/media-pool.js

```diff
@@ -60,6 +123,63 @@ const DOM_MEDIA_ELEMENT_ID_PREFIX = 'i-amphtml-media-';
 const POOL_MEDIA_ELEMENT_PROPERTY_NAME = '__AMP_MEDIA_POOL_ID__';
 
 
+/**
+ * @const {string}
+ */
+const ELEMENT_TASK_QUEUE_PROPERTY_NAME = '__AMP_MEDIA_ELEMENT_TASKS__';
+
+
+/** @const @enum {string} */
+export const ElementTaskName = {
+  BLESS: 'bless',
+  LOAD: 'load',
+  MUTE: 'mute',
+  PAUSE: 'pause',
+  PLAY: 'play',
+  REWIND: 'rewind',
+  UNMUTE: 'unmute',
+};
+
+
+/**
+ * @const {!Object<string, !ElementTaskDef>}
+ */
+const ELEMENT_TASKS = {
+  [ElementTaskName.BLESS]: el => blessMediaElement(el),
+  [ElementTaskName.LOAD]: el => Promise.resolve(el.load()),
+  [ElementTaskName.MUTE]: el => Promise.resolve(() => {
+    el.muted = true;
+    el.setAttribute('muted', '');
+  }),
+  [ElementTaskName.PAUSE]: el => Promise.resolve(el.pause()),
+  [ElementTaskName.PLAY]: el => {
+    if (!el.paused) {
+      // We do not want to invoke play() if the media element is already
+      // playing, as this can interrupt playback in some browsers.
+      return Promise.resolve();
+    }
+
+    // The play() invocation is wrapped in a Promise.resolve(...) due to the
+    // fact that some browsers return a promise from media elements' play()
+    // function, while others return a boolean.
+    return Promise.resolve(el.play());
+  },
+  [ElementTaskName.REWIND]: el => Promise.resolve(() => {
+    el.currentTime = 0;
+  }),
+  [ElementTaskName.UNMUTE]: el => Promise.resolve(() => {
+    el.muted = false;
+    el.removeAttribute('muted');
+  }),
+};
+
+
+/**
+ * @const {!ElementTaskDef}
+ */
+const NOOP_ELEMENT_TASK = () => Promise.resolve();
```

Comment:
Should this method store the expected ``taskName`` and log an error message saying this task does not exist?

Reply:
refactored this